doc/usage/extensions/autodoc.rst

```diff
@@ -662,6 +662,13 @@ There are also config values that you can set:
    .. __: https://mypy.readthedocs.io/en/latest/kinds_of_types.html#type-aliases
    .. versionadded:: 3.3
 
+.. confval:: autodoc_unqualified_typehints
+
+   If True, the leading module names of typehints of function signatures (ex. ``io.StringIO`` ->
+   `StringIO``).  Defaults to False.
```

Comment:
To keep compatibility, I'll disable this feature by default during 4.x. And I'd like to enable this since 5.x.